filewriter_v2/README.md

```diff
@@ -0,0 +1,387 @@
+
+#### SIMPLON V2 FileWriter Configuration Parameters
+
+The configuration interface is accessible at `http://<ADDRESS_OF_DCU>/api/v2/daq/config`.
+
+| Parameter | JSON Type | Default | Description |
+|-----------|------|---------|-------------|
+| `enabled` | boolean | `false` | Enabled state of the FileWriter. |
+| `image_id_offset` | uint | `1` | First image number on the axis `/entry/image_number`. |
+| `name_pattern` | string | `series_$id` | Base name of the files. `$id` will be replaced by the series number, i.e. using the default, the generated files will have the names:<br> `series_<series_number>_master.h5`<br> `series_<series_number>_data_<file_number>.h5`|
+| `images_per_file` | uint | `1000` | Maximum number of images stored in each data file. If set to `0`, all images are stored directly in the master file and no data files are created. <br> If set to a value greater than zero, the images are stored in multiple data files, but are still accessible in the master file via the dataset "/entry/data/data", since this is a [Virtual Dataset](https://portal.hdfgroup.org/display/HDF5/Virtual+Dataset++-+VDS) in this case.|
+| `v2_preview_enabled` | bool | `false` | Enabled state of the FileWriter V2 preview. |
+-->
+
+
+### Example
+
+FileWriter V2 may be configured with cURL.
+
+Using SIMPLON:
+```sh
+curl \
+    -X PUT \
+    -H "Content-Type: application/json" \
+    -d '{"mode": {"value": "enabled"}, "v2_preview_enabled": {"value": true}}' \
+    $ADDRESS_OF_DCU/filewriter/api/1.8.0/config
+```
+
+## Data Access
+
+The files are stored locally on the DCU and have to be transferred to the user computer, this can also be done via cURL using the SIMPLON API.
+
+```sh
+# list available files
+curl $ADDRESS_OF_DCU/filewriter/api/1.8.0/status/files
+
+# file access
+curl -O $ADDRESS_OF_DCU/data/<name_pattern>_master.h5
+curl -O $ADDRESS_OF_DCU/data/<name_pattern>_data_<file_number>.h5
+```
+
+For further details on available commands and status parameters please refer to the [SIMPLON API documentation](https://media.dectris.com/210607-DECTRIS-SIMPLON-API-Manual_EIGER2-chip-based_detectros.pdf).
+
+## Multi-channel Data
+
+One of the main new features of FileWriter V2 is the support of multi-channel data, i.e. each image consists of multiple channels, one per enabled threshold/difference mode. Therefore, the dataset `/entry/data/data` is four-dimensional, i.e. `[number_of_images, number_of_channels, width, height]`. The channels are identified via strings given in the [_AXISNAME_ field](https://manual.nexusformat.org/classes/base_classes/NXdata.html#nxdata-axisname-field) `channel` of the _NXdata_ class[^2]. For every channel name listed in this field, there exists a subgroup of the [_NXdetector_]() group of type _NXdetector_channel_[^1] with the same name followed by the suffix `_channel`. This group contains channel-specific metadata, for example `threshold_energy`, `flatfield` or `pixel_mask`.
```

Comment:
Thanks, using the verbatim style only for existing names is a very good point. How about this?

Acquired images may consist of multiple channels, one per enabled threshold/difference mode. FileWriter V2 introduces support for multi-channel image data. The dataset `/entry/data/data` is four-dimensional with shape $[n, m, h, w]$, where $n, m, h$ and $w$ denote the number of images, number of channels, height and width, respectively.

(It seems like I'm not able to find something that I'm happy with, so I'll aim for being content ;) )

Regarding the "AXISNAME field `channel`", I agree as well. I was shifting this sentence around so many times and it just would not sound good, so this is what I settled with :grimacing: Would it be better like this?
The channels are identified via strings given in the `channel` [axis](https://manual.nexusformat.org/classes/base_classes/NXdata.html#nxdata-axisname-field)  of the _NXdata_ class[^2].

Reply:
I was also not in love with what I wrote, but I got bored.

I think the proposed mathematical notation is a bit problematic. The NXmx documentation uses a different notation (e.g. nP, i, j, k, m). Particularly, their use of `m` for channel is also a different sort of channel, they seem to use `m` for a chromatic channel of the incident beam (see NXmx documentation for `incident_wavelength`).

I think your updated text about the axis is an improvement.

Reply:
I see your point, that could be confusing with the $m$. How about this?

The dataset `/entry/data/data` has the four dimensions _number of images, number of channels, height, width_ (ordered from slowest to fastest changing direction).

Reply:
I realized I used notation like `nP, i, j` in the tables below already, so I added a suggestion to use this for the shape: [aad30da](https://github.com/soph-dec/documentation/pull/1/commits/aad30da4a31a418bd77bdce64f1850d1a4bdecde) and [b4d6551](https://github.com/soph-dec/documentation/pull/1/commits/b4d655156e7642e153952057ffeb3be817ecd0f3).
what do you think?

Reply:
Looks good. I think using their notation is a reasonable and will probably avoid some confusion (even if i am not in love with their notation). `nC` for number of channels seems like a good fit.

Reply:
There is still a remaining inconsistency though. Now both `i` and `j` and `height` and `width` are used (the "Example" section uses `nP` and `height` and `width`). It would be more consistent to use `i` and `j` everywhere, however, admittedly I think this would be less readable. Another possibility would be to use another notation for `i` and `j`, perhaps `nY` and `nX`? I don't know.

Reply:
Hmm, good catch. I think I would go with the `i` and `j` notation, even though I don't like that notation very much as well.

Reply:
As discussed, I used `i, j` everywhere now, see [6a5efd2](https://github.com/soph-dec/documentation/pull/1/commits/6a5efd24d4609f68248e711544b92c54d40b0a0f).